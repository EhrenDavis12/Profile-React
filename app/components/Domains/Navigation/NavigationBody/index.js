/**
 *
 * Navigation
 *
 */

import React from 'react';
import PropTypes from 'prop-types';
// import styled from 'styled-components';
// import { Link } from 'react-router-dom';
import AppBar from 'components/Domains/Navigation/AppBar';
import Drawer from 'components/Domains/SharedKernel/Drawer';
import './index.css';

function NavigationBody({
  navigationLinks,
  toggleDrawer,
  selectItem,
  isDrawerOpen,
}) {
  /* const nav = (
    <nav>
      <ul>
        <li>
          <Link to="/">About Me</Link>
        </li>
      </ul>
    </nav>
  ); */
  return (
    <>
      <AppBar toggleDrawer={toggleDrawer} />
      <Drawer
        items={navigationLinks}
        selectItem={selectItem}
        itemLabelAttr="DisplayName"
        itemKeyAttr="key"
        isDrawerOpen={isDrawerOpen}
      />
    </>
  );
}

NavigationBody.propTypes = {
  // links: PropTypes.array.isRequired,
  navigationLinks: PropTypes.array.isRequired,
  toggleDrawer: PropTypes.func.isRequired,
  selectItem: PropTypes.func.isRequired,
  isDrawerOpen: PropTypes.bool.isRequired,
};

export default NavigationBody;
