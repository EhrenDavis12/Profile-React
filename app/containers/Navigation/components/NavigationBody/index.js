/**
 *
 * Navigation
 *
 */

import React from 'react';
import PropTypes from 'prop-types';
// import styled from 'styled-components';
// import { Link } from 'react-router-dom';
import NavTopButtons from 'containers/Navigation/components/NavTopButtons';
import Drawer from 'components/Drawer';

function NavigationBody({
  navigationLinks,
  toggleDrawer,
  selectItem,
  isDrawerOpen,
}) {
  return (
    <React.Fragment>
      <NavTopButtons
        toggleDrawer={toggleDrawer}
        items={navigationLinks}
        selectItem={selectItem}
        itemLabelAttr="DisplayName"
        itemKeyAttr="key"
      />
      <Drawer
        items={navigationLinks}
        selectItem={selectItem}
        itemLabelAttr="DisplayName"
        itemKeyAttr="key"
        isDrawerOpen={isDrawerOpen}
      />
    </React.Fragment>
  );
}

NavigationBody.propTypes = {
  navigationLinks: PropTypes.array.isRequired,
  toggleDrawer: PropTypes.func.isRequired,
  selectItem: PropTypes.func.isRequired,
  isDrawerOpen: PropTypes.bool.isRequired,
};

export default NavigationBody;
