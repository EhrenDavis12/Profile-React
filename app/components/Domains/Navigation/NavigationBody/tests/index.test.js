import React from 'react';
import renderer from 'react-test-renderer';
// import { mount } from 'enzyme';
// import { enzymeFind } from 'styled-components/test-utils';

import NavigationBody from '../index';

describe('<NavigationBody />', () => {
  it('renders as expected', () => {
    const tree = renderer
      .create(
        <NavigationBody
          navigationLinks={[
            {
              key: 'AboutMe',
              LinkTo: '/',
              DisplayName: 'About Me',
            },
            {
              key: 'Portfolio',
              LinkTo: '/portfolio',
              DisplayName: 'Portfolio',
            },
            {
              key: 'Contact',
              LinkTo: '/contact',
              DisplayName: 'Contact',
            },
            {
              key: 'Messages',
              LinkTo: '/messages',
              DisplayName: 'Messages',
            },
          ]}
          toggleDrawer={() => {}}
          selectItem={() => {}}
          isDrawerOpen={false}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
