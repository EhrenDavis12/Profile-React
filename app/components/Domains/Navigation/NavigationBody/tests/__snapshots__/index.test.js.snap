// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavigationBody /> renders as expected 1`] = `
Array [
  <div
    className="appBar"
  >
    <div
      className="iconButton"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
      tabIndex={0}
    >
      <span
        aria-hidden={true}
        className="fa fa-bars icon"
      />
    </div>
    <div
      className="heading"
    >
      Coder Daily
    </div>
    <div
      className="linkContainer"
    >
      Log In
    </div>
  </div>,
  <div
    className="drawer"
  >
    <div
      className="item"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
      tabIndex={0}
    >
      About Me
    </div>
    <div
      className="item"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
      tabIndex={0}
    >
      Portfolio
    </div>
    <div
      className="item"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
      tabIndex={0}
    >
      Contact
    </div>
    <div
      className="item"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
      tabIndex={0}
    >
      Messages
    </div>
  </div>,
]
`;
